src/Chapter3.hs

```diff
@@ -374,6 +386,31 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Creature = Creature
+  {
+    health :: Int,
+    attack :: Int,
+    gold :: Int
+  }
+
+
+-- turn 1 is Knight, 0 is Monster
+fight :: Creature -> Creature -> Int -> Int
```

Comment:
Nice catch on noticing that both knight and monster looks the same data type wise! 👍🏼 
The only issue with reusage of the same data type for Monster and Knight here is that you can easily mix them when calling the function:

```haskell
> fight knight monster
> fight monster knight
```

Both of the calls are absolutely valid, but the results are going to be different!

Here, we had in mind to make two types for each Monster and Knight. But as you already noticed, these two would be almost identical, and it feels wrong to duplicate that. Later we show the `newtypes` which usually resolve these issues (we didn't expect you to use them in here, this is just the additional note 🙂 )
```haskell
newtype Monster = Monster Entity
newtype Knight = Knight Entity
```